src/main/java/jenkins/scm/api/SCMFileSystem.java

```diff
@@ -500,8 +522,11 @@ public final boolean supports(SCMSourceDescriptor descriptor) {
          * @throws InterruptedException if the attempt to create a {@link SCMFileSystem} was interrupted.
          */
         @CheckForNull
-        public abstract SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull SCMRevision rev)
-                throws IOException, InterruptedException;
+        public SCMFileSystem build(@NonNull Item owner, @NonNull SCM scm, @CheckForNull SCMRevision rev)
+                throws IOException, InterruptedException
+        {
+            throw new AbstractMethodError("Implement build(Item, SCM, SCMRevision, Run)");
```

Comment:
nit: Maybe this would already be clear to developers, but this API may still be called directly and so both overloads need to be implemented.

```suggestion
            throw new AbstractMethodError("Implement build(Item, SCM, SCMRevision, Run) and override this method to delegate to it");
```